lib/ecto/adapters/myxql/connection.ex

```diff
@@ -687,6 +687,17 @@ if Code.ensure_loaded?(MyXQL) do
       error!(query, "MySQL adapter does not support aggregate filters")
     end
 
+    defp expr({:top_level, %Ecto.SubQuery{query: query}}, sources, parent_query) do
+      combinations =
+        Enum.map(query.combinations, fn {type, combination_query} ->
+          {type, put_in(combination_query.aliases[@parent_as], {parent_query, sources})}
+        end)
+
+      query = put_in(query.combinations, combinations)
+      query = put_in(query.aliases[@parent_as], {parent_query, sources})
+      [all(query, subquery_as_prefix(sources))]
+    end
```

Comment:
There is some duplication with the function below, but you can implement the function below by calling `[?(, top_level_expr(query, ...), ?)]`.